syntax = "proto2";
package DB.Protos;

message DataPartModel
{
    required string type = 1;
    required uint64 disk_id = 2;
    required uint64 bytes_on_disk = 3;
    required uint64 rows_count = 4;     /// same as what in count.txt
    required string partition_and_minmax = 5;
    required uint64 last_modify_time = 6;
    optional string uuid = 7;           /// optional uuid for parts; set only if assign_part_uuids is configured in storage.
    optional string columns = 8;
    optional uint64 meta_version = 9;   /// used for replaceing columns in later optimization. 
    optional string ttl_infos = 10;
    optional uint64 marks_count = 11;    /// save marks_count for non-adaptive parts;
    repeated uint64 index_granularities = 12;  /// defails of adaptive index granularity
    optional uint64 index_offset = 13;   /// index offset in hdfs meta file. 0 for local data part
    optional string versions = 14;       /// serialized MergeTreeDataPartVersions
    optional string parent_part = 15;    /// parent part name for projections.
    optional string compress_codec = 16;
};